setup/gcp/installer/util/gcloud.py

```diff
@@ -0,0 +1,786 @@
+
+
+def enable_os_login(instance_name, zone):
+    """Enable os login for the given VM instance.
+
+    Args:
+        instance_name (str): Name of the VM instance
+        zone (str): Zone of the VM instance
+    """
+    gcloud_command_args = ['gcloud', 'compute', 'instances', 'add-metadata',
+                           instance_name, '--metadata', 'enable-oslogin=TRUE',
+                           '--zone', zone]
+
+    return_code, _, err = run_command(gcloud_command_args)
+    if return_code:
+        print (err)
+
+
+def create_deployment(project_id,
+                      organization_id,
+                      deploy_tpl_path,
+                      template_type,
+                      datetimestamp,
+                      dry_run):
+    """Create the GCP deployment.
+
+    Args:
+        project_id (str): GCP project id
+        organization_id (str): GCP organization id
+        deploy_tpl_path (str): Path of deployment template
+        template_type (str): Type of the template (client/server)
+        datetimestamp (str): Timestamp
+        dry_run (bool): Whether the installer is in dry run mode
+
+    Returns:
+        str: Name of the deployment
+        int: The return code value of running `gcloud` command to create
+            the deployment.
+    """
+
+    def _ping_deployment_manager():
+        """Check deployment manager status.
+        """
+        run_command(
+            ['gcloud', 'deployment-manager', 'deployments',
+             'describe', 'testing-deployment-manager-connection'])
+
+    print_banner('Create Forseti deployment')
+
+    if dry_run:
+        print('This is a dry run, so skipping this step.')
+        return 0
+
+    print ('This may take a few minutes.')
+    _ping_deployment_manager() # Make sure deployment-manager is ready
+    deployment_name = 'forseti-security-{}-{}'.format(template_type,
+                                                      datetimestamp)
+    print('Deployment name: {}'.format(deployment_name))
+    print('Deployment Manager Dashboard: '
+          'https://console.cloud.google.com/deployments/details/'
+          '{}?project={}&organizationId={}\n'.format(
+              deployment_name, project_id, organization_id))
+    return_code, out, err = run_command(
+        ['gcloud', 'deployment-manager', 'deployments', 'create',
+         deployment_name, '--config={}'.format(deploy_tpl_path)])
+    if return_code:
+        print(err)
+    else:
+        print(out)
+        print('\nCreated deployment successfully.')
+
+    return deployment_name, return_code
+
+
+def check_vm_init_status(vm_name, zone):
+    """Check vm initialization status.
+
+    Args:
+        vm_name (str): Name of the VM instance.
+        zone (str): Zone of the VM instance.
+
+    Returns:
+        bool: Whether or not the VM has finished initializing.
+    """
+
+    check_script_executed = 'tail -n1 /tmp/deployment.log'
+
+    _, out, _ = run_command(
+        ['gcloud', 'compute', 'ssh', vm_name,
```

Comment:
Open question, can this be done by watching the stackdriver logs using the log API instead of sshing into the VM?

What about by watching the console output of the VM? https://cloud.google.com/compute/docs/reference/rest/v1/instances/getSerialPortOutput

Reply:
Opened an issue for this, we can do more research on this topic later on. #1112